src/components/ai-input.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useRef, useEffect } from 'react';
+import Button from './button';
+import { AssistantIcon } from './icons/assistant';
+import { MenuIcon } from './icons/menu';
+import { ReturnIcon } from './icons/return';
+
+interface AIInputProps {
+	disabled: boolean;
+	input: string;
+	setInput: ( input: string ) => void;
+	handleSend: () => void;
+	handleKeyDown: ( e: React.KeyboardEvent< HTMLTextAreaElement > ) => void;
+	clearInput: () => void;
+	isAssistantThinking: boolean;
+}
+
+const MAX_ROWS = 10;
+
+export const AIInput: React.FC< AIInputProps > = ( {
+	disabled,
+	input,
+	setInput,
+	handleSend,
+	handleKeyDown,
+	clearInput,
+	isAssistantThinking,
+} ) => {
+	const inputRef = useRef< HTMLTextAreaElement >( null );
+
+	useEffect( () => {
+		if ( ! disabled && inputRef.current ) {
+			inputRef.current.focus();
+		}
+	}, [ disabled ] );
+
+	const handleInput = ( e: React.ChangeEvent< HTMLTextAreaElement > ) => {
+		setInput( e.target.value );
+
+		if ( inputRef.current ) {
+			// Reset the height of the textarea to auto to recalculate the height
+			inputRef.current.style.height = 'auto';
+
+			// Calculate the maximum height based on the maximum number of rows
+			const lineHeight = parseInt( window.getComputedStyle( inputRef.current ).lineHeight, 10 );
+			const maxHeight = MAX_ROWS * lineHeight;
+
+			// Set the height of the textarea to the minimum of its scrollHeight and the maximum height
+			inputRef.current.style.height = `${ Math.min( inputRef.current.scrollHeight, maxHeight ) }px`;
+
+			// If the scrollHeight exceeds the maximum height, enable vertical scrolling
+			// Otherwise, hide the scrollbar
+			inputRef.current.style.overflowY =
+				inputRef.current.scrollHeight > maxHeight ? 'auto' : 'hidden';
+
+			// Scroll to the bottom if the content exceeds the maximum height
+			inputRef.current.scrollTop = inputRef.current.scrollHeight;
```

Comment:
I did not note any performance degradation or spend much time measuring, but I wonder if there are any efficiencies we might make here. 

* Are we able to cache `lineHeight` outside of the event handler? 
* Should we perform all "writes" together? I.e., mixing "reading" and "writing" from the DOM [might result](https://dev.to/gopal1996/understanding-reflow-and-repaint-in-the-browser-1jbg) multiple, unnecessary repaints. 


The answer may be "no, there are not efficiencies we should pursue," but I figured I'd share the thought regardless.

Reply:
Thank you for the feedback here. Working on managing the height of this textarea actually comprised the majority of the time spent on this component, so I definitely appreciate a second perspective on the approach.

I really was trying for a "natural" solution where the `height` attribute did not need to be managed via Javascript, and could be managed by flexbox properties and lineHeight alone, but ended up timeboxing it. I do think it is worth pursuing optimizations in a follow-up PR.